bundles/io/org.eclipse.smarthome.io.transport.mqtt.test/src/test/java/org/eclipse/smarthome/io/transport/mqtt/test/MqttServiceTests.java

```diff
@@ -0,0 +1,117 @@
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.smarthome.io.transport.mqtt.test;
+
+import static org.junit.Assert.*;
+
+import java.util.Dictionary;
+import java.util.Hashtable;
+import java.util.Map;
+
+import javax.naming.ConfigurationException;
+
+import org.eclipse.paho.client.mqttv3.MqttException;
+import org.eclipse.smarthome.io.transport.mqtt.MqttBrokerConnection;
+import org.eclipse.smarthome.io.transport.mqtt.MqttBrokersObserver;
+import org.eclipse.smarthome.io.transport.mqtt.MqttService;
+import org.junit.Test;
+
+/**
+ * Tests the MqttService class
+ *
+ * @author David Graeff - Initial contribution
+ */
+public class MqttServiceTests {
+    boolean brokerExisting = false;
+
+    // Tests addBrokersListener/removeBrokersListener
+    @Test
+    public void brokerConnectionListenerTests() {
+        MqttService service = new MqttService();
+        assertFalse(service.isBrokerObservers());
+        MqttBrokersObserver observer = new MqttBrokersObserver() {
+            @Override
+            public void brokerAdded(MqttBrokerConnection broker) {
+                brokerExisting = true;
+            }
+
+            @Override
+            public void brokerRemoved(MqttBrokerConnection broker) {
+                brokerExisting = false;
+            }
+        };
+        service.addBrokersListener(observer);
+        assertTrue(service.isBrokerObservers());
+
+        MqttBrokerConnection connection;
+        try {
+            connection = new MqttBrokerConnection("name", "tcp://123.123.123.123", false);
+        } catch (ConfigurationException c) {
+            fail("Couldn't create a MqttBrokerConnection object");
+            return;
+        }
+        assertTrue(service.addBrokerConnection(connection));
+        assertTrue(brokerExisting);
+
+        service.removeBrokerConnection(connection);
+        assertFalse(brokerExisting);
+
+        service.removeBrokersListener(observer);
+        assertFalse(service.isBrokerObservers());
+    }
+
+    // Tests extractBrokerConfigurations() and addBrokerConnection(map)
+    @Test
+    public void textualConfigurationTests() {
+        MqttService service = new MqttService();
+
+        Dictionary<String, String> properties = new Hashtable<>();
+        properties.put("bam.name", "brokername");
+        properties.put("bam.url", "tcp://123.123.123.123");
+        Map<String, Map<String, Object>> map = service.extractBrokerConfigurations(properties);
+        assertNotNull(map);
+        assertEquals(map.size(), 1);
+        Map<String, Object> data = map.get("bam");
+        assertNotNull(data);
+        assertTrue("brokername".equals(data.get("name")));
+        assertTrue("tcp://123.123.123.123".equals(data.get("url")));
+
+        assertEquals(service.getAllBrokerConnections().size(), 0);
+        try {
+            service.addBrokerConnection(data);
+        } catch (ConfigurationException | MqttException e) {
```

Comment:
same as in the `MqttBrokerConnectionTests`. You dont want to test the Exception here. Just let it fail with its own Exception.